### PR TITLE
fix: constexpr bit_cast with empty base classes

### PR DESCRIPTION
**Pull request by sethp**
Prior to this commit, clang would fail to produce a constant value for `b` in:

```c++
struct base {
};

struct s : base {
    int z;
};

constexpr auto b = std::bit_cast<s>(0x12); 
```

e.g. https://godbolt.org/z/srrbTMPq4

### CODE REVIEW

**Comment by sethp** [comment]
Just to cross-link, I originally asked about this on discord: https://discord.com/channels/636084430946959380/636732781086638081/1209534556126973993 

(cc @cor3ntin )

**Comment by cor3ntin** [comment]
This will need a release note.
I'm curious why the removed code was there in the first place. Any idea @tbaederr ?

**Review by cor3ntin — APPROVED** [review]
LGTM, thanks

File: clang/docs/ReleaseNotes.rst [hunk 1]
```diff
@@ -280,6 +280,8 @@ Bug Fixes to C++ Support
   was only accepted at namespace scope but not at local function scope.
 - Clang no longer tries to call consteval constructors at runtime when they appear in a member initializer.
   (`#782154 <https://github.com/llvm/llvm-project/issues/82154>`_`)
```
Comment (sethp) [inline clang/docs/ReleaseNotes.rst:282]: I think this should be:

```suggestion
  (`#82154 <https://github.com/llvm/llvm-project/issues/82154>`_`)
```

Which would be easy to fix as part of this PR, or a new one?

Comment (cor3ntin) [reply clang/docs/ReleaseNotes.rst:282]: Oups. New PR (Or nfc change, no need for a review for that)

**Comment by sethp** [comment]
@cor3ntin @tbaederr is there anything else you need from me here? I would expect github's "squash and merge" button to work fine for this change, but I could rebase if you'd prefer.

**Comment by tbaederr** [comment]
There's still a merge conflict now. Do you not have the necessary permissions to merge it yourself?

**Comment by sethp** [comment]
Ah, got it, thank you: I am once again wishing for a `.gitattributes` "keep both" merge strategy more usable than `union` for the release notes.

> Do you not have the necessary permissions to merge it yourself?

No, I don't have write permissions on the LLVM repo: I think I will need one of y'all's help hitting the button here. I'll look into getting permissions, too, but I presume there's some steps involved.

**Pull request merged**